src/AudioDeviceTestWidget/AudioDeviceTestWidget.test.tsx

```diff
@@ -0,0 +1,196 @@
+import { mount, shallow } from 'enzyme';
+
+import Alert from '../common/Alert/Alert';
+import AudioDevice from './AudioDevice/AudioDevice';
+import AudioDeviceTestWidget from './AudioDeviceTestWidget';
+import useTestRunner from './useTestRunner/useTestRunner';
+
+jest.mock('./AudioDevice/AudioDevice');
+jest.mock('./useTestRunner/useTestRunner');
+const mockAudioDevice = AudioDevice as jest.Mock<any>;
+const mockUseTestRunner = useTestRunner as jest.Mock<any>;
+
+describe('the AudioDeviceTestWidget component', () => {
+  let hookProps: any;
+
+  beforeEach(() => {
+    hookProps = {
+      error: '',
+      inputLevel: 0,
+      isRecording: false,
+      isAudioInputTestRunning: false,
+      isAudioOutputTestRunning: false,
+      outputLevel: 0,
+      playAudio: jest.fn(),
+      playbackURI: '',
+      readAudioInput: jest.fn(),
+      testEnded: false,
+    };
+    mockUseTestRunner.mockImplementation(() => hookProps);
+    mockAudioDevice.mockImplementation(() => null);
+  });
+
+  it('should render correct components on load', () => {
+    const wrapper = shallow(<AudioDeviceTestWidget/>);
+    expect(wrapper.find(AudioDevice).length).toEqual(2);
+
+    const outputDevice = wrapper.find(AudioDevice).at(0);
+    const inputDevice = wrapper.find(AudioDevice).at(1);
+    const recordBtn = wrapper.find(Button).at(0);
+    const playBtn = wrapper.find(Button).at(1);
+
+    expect(wrapper.find(Alert).exists()).toBeFalsy();
+    expect(outputDevice.prop('disabled')).toBeFalsy();
+    expect(inputDevice.prop('disabled')).toBeFalsy();
+    expect(recordBtn.prop('disabled')).toBeFalsy();
+    expect(playBtn.prop('disabled')).toBeTruthy();
+    expect(recordBtn.text()).toEqual('Record');
+    expect(playBtn.text()).toEqual('Play');
+  });
+
+  describe('passive testing', () => {
+    it('should start passive testing by default', () => {
+      mount(<AudioDeviceTestWidget/>);
+      expect(hookProps.readAudioInput).toHaveBeenCalledWith({ deviceId: '' });
+    });
+
+    [
+      {
+        isCalled: true, props: { error: '', isRecording: false, isAudioInputTestRunning: false }
+      },{
+        isCalled: false, props: { error: '', isRecording: true, isAudioInputTestRunning: false }
+      },{
+        isCalled: false, props: { error: '', isRecording: false, isAudioInputTestRunning: true }
+      },{
+        isCalled: false, props: { error: '', isRecording: true, isAudioInputTestRunning: true }
+      },{
+        isCalled: false, props: { error: 'foo', isRecording: false, isAudioInputTestRunning: false }
+      },{
+        isCalled: false, props: { error: 'foo', isRecording: true, isAudioInputTestRunning: false }
+      },{
+        isCalled: false, props: { error: 'foo', isRecording: false, isAudioInputTestRunning: true }
+      },{
+        isCalled: false, props: { error: 'foo', isRecording: true, isAudioInputTestRunning: true }
+      }
+    ].forEach(({isCalled, props}) => {
+      it(`when props are ${JSON.stringify(props)}`, () => {
```

Comment:
It isn't clear what this is testing when looking at the verbiage alone:

![image](https://user-images.githubusercontent.com/12685223/100953802-84c0c580-34d0-11eb-89a7-b475d86b7b78.png)

What about something like this?

```suggestion
      it(`should${isCalled ? ' ' : ' not '}call readAudioInput when props are ${JSON.stringify(props)}`, () => {
```
![image](https://user-images.githubusercontent.com/12685223/100953956-d9644080-34d0-11eb-8095-3b7020174e5b.png)